src/Omnipay/Common/HasParametersTrait.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Omnipay\Common;
+
+trait HasParametersTrait
+{
+    /**
+     * Internal storage of all of the parameters.
+     *
+     * @var ParameterBag
+     */
+    protected $parameters;
```

Comment:
It's not currently used outside of HasParametersTrait but making it private ensures that it can't be used outside of that trait (e.g. in any use-ing class) and can't be used in any trait that extends that trait. So I would prefer it remain protected because you can't see into the future and know that nobody will want to extend this trait.